### ISSUE TITLE
[no-unsafe-member-access] Extending namespaces trigger error

### ISSUE THREAD

**Issue by susnux** [typescript-eslint/typescript-eslint#2740]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
```TS
declare namespace FG {
	interface A {}
}

interface B extends FG.A {
}
```
You can also look at this example repo: https://github.com/susnux/typescript-eslint-no-unsafe-member-access

**Expected Result**
No error message

**Actual Result**

> typescript-eslint-no-unsafe-member-access/src/test.d.ts
>  6:21  error  Unsafe member access .A on an any value  @typescript-eslint/no-unsafe-member-access

**Additional Info**
Nothing special, you can simply reproduce this.
I experienced the same issue (this) as reported by @tao-cumplido in #1868.

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.27.0` |
| `@typescript-eslint/parser`             | `2.27.0 |
| `@typescript-eslint/typescript-estree`  | `2.27.0` |
| `@typescript-eslint/experimental-utils` | `2.27.0` |
| `TypeScript`                            | `4.0.5` |
| `node`                                  | `12.18.4` |
| `npm`                                   | `6.14.6` |

**Comment by bradzacher** [comment]
The root of this problem is that for `TSInterfaceHeritage` we emit a `MemberExpression` for the heritage.

```ts
class B extends FG.A {} // FG.A is a MemberExpression ✅
type B = FG.A; // FG.A is a TSQualifiedName ✅

class B implements FG.A {} // FG.A is a MemberExpression ❌
interface B extends FG.A {} // FG.A is a MemberExpression ❌
```
https://astexplorer.net/#/gist/4d80b0e52f9c5a85c3e8e5965159c0b0/f19196f6c60b81e792172e7c0adcfc8f676270d0

However - this is a type-only location, so _ideally_ it should be a `TSQualifiedName`.

### Why do we emit a `MemberExpression`?
Because TS emits a `PropertyAccessExpression` here - and our parser (mostly) just converts TS's AST as written.

### Why does TS emit this?
Because the TS parser shares the logic for parsing this between:
- `class` `extends`
- `interface` `extends`
- `class` `implements`

TS has a _semantic_ error to prevent you from doing anything too stupid - however it does allow some stupid code: https://github.com/microsoft/TypeScript/issues/41420

### What's the fix?

Pretty straight-forward - the selectors need to be updated to ignore any `MemberExpression` whose parent is either `TSInterfaceHeritage` or `TSClassImplements`.

https://github.com/typescript-eslint/typescript-eslint/blob/e82698c0ec796e460e40c3dd90a30bd100db05c2/packages/eslint-plugin/src/rules/no-unsafe-member-access.ts#L74-L75
